src/BindPattern.jl

```diff
@@ -25,26 +25,93 @@ struct BinderState
     # A dictionary used to intern CodePoint values in Match2Cases.
     intern::Dict
 
+    # A counter used to dispense unique integers to make prettier gensyms
+    gensyms::Vector{Symbol}
+
     function BinderState(mod::Module, input_variable::Symbol)
         new(
             mod,
             input_variable,
             Dict{BoundFetchPattern, Symbol}(),
             Vector{Pair{LineNumberNode, String}}(),
             Vector{Any}(),
-            Dict()
+            Dict(),
+            Symbol[]
         )
     end
 end
 
+function gensym(base::String, state::BinderState)::Symbol
+    s = gensym("$(base)_$(length(state.gensyms))")
+    push!(state.gensyms, s)
+    s
+end
+gensym(base::String) = Base.gensym(base)
+
+#
+# Generate a fresh synthetic variable whose name hints at its purpose.
+#
+function gentemp(p::BoundFetchPattern)
+    error("not implemented: gentemp(::$(typeof(p)))")
+end
+function gentemp(p::BoundFetchFieldPattern)
```

Comment:
does this need a method for `BoundFetchExpressionPattern`?

Reply:
No.  These functions are only called from `get_temp`, just below, and there is special handling for that type that doesn't call this function.